circuits/cpp/barretenberg/cpp/src/barretenberg/honk/flavor/ecc_vm.hpp

```diff
@@ -0,0 +1,859 @@
+     * circuits.
+     */
+    using VerificationKey = VerificationKey_<PrecomputedEntities<Commitment, CommitmentHandle>>;
+
+    /**
+     * @brief A container for polynomials handles; only stores spans.
+     */
+    using ProverPolynomials = AllEntities<PolynomialHandle, PolynomialHandle>;
+
+    /**
+     * @brief A container for polynomials produced after the first round of sumcheck.
+     * @todo TODO(#394) Use polynomial classes for guaranteed memory alignment.
+     */
+    using FoldedPolynomials = AllEntities<std::vector<FF>, PolynomialHandle>;
+
+    using RawPolynomials = AllEntities<Polynomial, PolynomialHandle>;
+
+    /**
+     * @brief A container for polynomials produced after the first round of sumcheck.
+     * @todo TODO(#394) Use polynomial classes for guaranteed memory alignment.
+     */
+    using RowPolynomials = AllEntities<FF, FF>;
+
+    /**
+     * @brief A container for storing the partially evaluated multivariates produced by sumcheck.
+     */
+    class PartiallyEvaluatedMultivariates : public AllEntities<Polynomial, PolynomialHandle> {
+
+      public:
+        PartiallyEvaluatedMultivariates() = default;
+        PartiallyEvaluatedMultivariates(const size_t circuit_size)
+        {
+            // Storage is only needed after the first partial evaluation, hence polynomials of size (n / 2)
+            for (auto& poly : this->_data) {
+                poly = Polynomial(circuit_size / 2);
+            }
+        }
+    };
+
+    /**
+     * @brief A container for univariates produced during the hot loop in sumcheck.
+     * @todo TODO(#390): Simplify this by moving MAX_RELATION_LENGTH?
+     */
+    template <size_t MAX_RELATION_LENGTH>
+    using ExtendedEdges =
+        AllEntities<sumcheck::Univariate<FF, MAX_RELATION_LENGTH>, sumcheck::Univariate<FF, MAX_RELATION_LENGTH>>;
+
+    /**
+     * @brief A container for the polynomials evaluations produced during sumcheck, which are purported to be the
+     * evaluations of polynomials committed in earlier rounds.
+     */
+    class ClaimedEvaluations : public AllEntities<FF, FF> {
+      public:
+        using Base = AllEntities<FF, FF>;
+        using Base::Base;
+        ClaimedEvaluations(std::array<FF, NUM_ALL_ENTITIES> _data_in) { this->_data = _data_in; }
+    };
+
+    /**
+     * @brief A container for commitment labels.
+     * @note It's debatable whether this should inherit from AllEntities. since most entries are not strictly needed. It
+     * has, however, been useful during debugging to have these labels available.
+     *
+     */
+    class CommitmentLabels : public AllEntities<std::string, std::string> {
+      private:
+        using Base = AllEntities<std::string, std::string>;
+
+      public:
+        CommitmentLabels()
+            : AllEntities<std::string, std::string>()
+        {
+            Base::transcript_add = "transcript_add";
```

Comment:
Why are some labels lowercase and other uppercase?

Reply:
mistake, will fix